### DISCUSSION TITLE
Using SMTP with WHMCS

### DISCUSSION THREAD

**Discussion by stevennoad** [postalserver/postal#2357] [Help with using Postal]
Has anyone successfully configured WHMCS to send emails via postal? When enabling SMTP within WHMCS, and entering the SMTP details postal show after creating a Credential this error is returned: Validation Error: SMTP connect() failed. 

I'm probably missing something really simply, any help would be appreciated. I've added details below:

WHMCS (version: 8.6.1):
- Mail Provider: SMTP
- Mail Encoding: 8bit
- Service Provider: Generic
- SMTP Host: postal.my-domain.co.uk [used what is shown in postal]
- SMTP Port: 25 [used what is shown in postal]
- SMPT Authentication: Password
- SMTP Username: organisation/server [used what is shown in postal]
- SMTP Password: my-password [used what is shown in postal]
- SMTP SSL Type: SSL

VPS is hosted on Katapult running Ubuntu 22.01. I have run telnet postal.my-domain.co.uk 25 which returns how I would expect it to return. I've checked ufw and ensured port 25 is open and allowed from anywhere. If I use the API method (or php library postal works as expected).

**Comment by willpower232** [comment]
Sounds like everything is in order, have you tried setting SSL off in WHMCS to see if that helps? If SSL isn't set up right in Postal, it can be difficult to debug.

You could also try `openssl s_client -connect postal.my-domain.co.uk:25 -starttls smtp` to help verify SSL is set up right.

**Reply by stevennoad** [reply]
I have tried setting SSL too off in WHMCS with the same results. Running that command in the terminal window gives me the below output:

```
CONNECTED(00000003)
Didn't find STARTTLS in server response, trying anyway...
40E7874E627F0000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../ssl/record/ssl3_record.c:354:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 132 bytes and written 357 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---
```

In terms of setting up postal, I just followed the steps in the docs using docker/docker-compose and caddy. I have tried a variety of different port numbers in the config/postal.yml with the same outcome.

```
smtp_server:
# Specify configuration to the Postal SMTP server
port: 25
```

**Reply by willpower232** [reply]
The openssl command confirms that you haven't currently got SMTP enabled so that setting should remain off unless you plan on enabling it.

The other thing you can try is checking the SMTP logs `postal logs smtp` to see what happens when WHMCS tries to connect. If you don't see any content then it isn't reaching your Postal at all and if you do, you may be able to see what the problem is.

**Reply by stevennoad** [reply]
Thanks Will for the info, it looks like it was me doing things wrong. 

After looking into it further I found when I was editing the postal.yml and then running `postal start` the changes are not reflected (I thought they were). As soon as I did `postal stop` then `postal start` it started working and now sending email via WHMCS via SSL/TLS is working as expected. Thanks again for your time.
